apps/condo/domains/acquiring/schema/RegisterMultiPaymentService.js

```diff
@@ -0,0 +1,231 @@
+                }
+                const receiptsIds = groupedReceipts.flatMap(group => group.receiptsIds)
+                const uniqueReceiptsIds = new Set(receiptsIds)
+                if (receiptsIds.length !== uniqueReceiptsIds.size) {
+                    throw new Error(REGISTER_MP_RECEIPTS_DUPLICATE)
+                }
+
+                // Stage 1. Check Acquiring
+                const consumers = await find('ServiceConsumer', {
+                    id_in: consumersIds,
+                })
+                if (consumers.length !== consumersIds.length) {
+                    const existingConsumerIds = consumers.map(consumer => consumer.id)
+                    const missingConsumerIds = consumersIds.filter(consumerId => !existingConsumerIds.includes(consumerId))
+                    throw new Error(`${REGISTER_MP_REAL_CONSUMER_MISMATCH} Missing: ${missingConsumerIds.join(', ')}`)
+                }
+                const contextMissingConsumers = consumers
+                    .filter(consumer => !get(consumer, 'acquiringIntegrationContext'))
+                    .map(consumer => consumer.id)
+                if (contextMissingConsumers.length) {
+                    throw new Error(`${REGISTER_MP_NO_ACQUIRING_CONSUMERS} (${contextMissingConsumers.join(', ')})`)
+                }
+
+                const consumersByIds = Object.assign({}, ...consumers.map(obj => ({ [obj.id]: obj })))
+
+                const uniqueAcquiringContextsIds = new Set(consumers.map(consumer => consumer.acquiringIntegrationContext))
+                const acquiringContexts = await find('AcquiringIntegrationContext', {
+                    id_in: Array.from(uniqueAcquiringContextsIds),
+                })
+
+                const acquiringContextsByIds = Object.assign({}, ...acquiringContexts.map(obj => ({ [obj.id]: obj })))
+
+                const acquiringIntegrations = new Set(acquiringContexts.map(context => context.integration))
+                if (acquiringIntegrations.size !== 1) {
+                    throw new Error(REGISTER_MP_MULTIPLE_INTEGRATIONS)
+                }
+
+                // NOTE: Here using serverSchema to get many relation
+                let acquiringIntegration = await AcquiringIntegration.getAll(context, {
+                    id: Array.from(acquiringIntegrations)[0],
+                })
+                acquiringIntegration = get(acquiringIntegration, '0')
+                // const acquiringIntegration = await getById('AcquiringIntegration', Array.from(acquiringIntegrations)[0])
+
+                // TODO (savelevMatthew): check that all receipts linked to right consumers?
+                // Stage 2. Check BillingReceipts
+                if (receiptsIds.length > 1 && !acquiringIntegration.canGroupReceipts) {
+                    throw new Error(REGISTER_MP_CANNOT_GROUP_RECEIPTS)
+                }
+                const receipts = await find('BillingReceipt', {
```

Comment:
deletedAt :null

Reply:
no....we're processing case of deletedAt explicitly to inform user...
```JavaScript
const deletedReceiptsIds = receipts.filter(receipt => Boolean(receipt.deletedAt)).map(receipt => receipt.id)
if (deletedReceiptsIds.length) {
    throw new Error(`${REGISTER_MP_DELETED_RECEIPTS} (${deletedReceiptsIds.join(', ')})`)
}
```